src/core/makefile.x/makefile.osx

```diff
@@ -1,7 +1,8 @@
 # uncomment the following to force 32-bit compilation
 # FORCE_M32=-m32
 
-ifneq ($(shell sw_vers -productVersion | egrep '10\.(6|7|8|9|10|11|12|13|14|15)(\.[0-9]+)?'),)
+# the sw_vers trick only works after 10.6
```

Comment:
recently pushed a different and conflicting change here and thus this change has been not include for now. if you'd like, please take a look in main branch and advise how to proceed. thanks!

Reply:
both fixes seem equivalent. no problem with the different fix from my POV.